### DISCUSSION TITLE
show .wasm loading progress

### DISCUSSION THREAD

**Discussion by iTrooz** [emscripten-core/emscripten#21222] [Q&A] [answered]
Hey ! I see some people online managed to show the loading percentage of the .wasm main module, but I didn't find any clean implementation (for example, https://github.com/WordPress/wordpress-playground/pull/46 works but seems to patch emscripten functions, making it prone to break in the future)

I wanted to know if there was an official, supported way to do this
There is also https://github.com/emscripten-core/emscripten/issues/16278 but it doesn't have any answer

Thanks !

**Comment by curiousdannii** [comment, marked as answer]
If you preload the .wasm file and then pass it to the module using `wasmBinary`, then you can download it with any sort of normal fetch progress code you like.

A general solution probably wouldn't make a lot of sense, as there's no consistent shared UI for Emscripten projects. Many only exist in the background, and have no control over any HTML elements.

**Reply by iTrooz** [reply]
Ok ! Thanks
